diff --git a/command-not-found.sh b/command-not-found.sh
index 0804957..132577a 100755
--- a/command-not-found.sh
+++ b/command-not-found.sh
@@ -41,28 +41,26 @@ command_not_found_handle () {
             if ! [ -z "$NIX_AUTO_INSTALL" ]; then
                 >&2 cat <<EOF
 The program '$cmd' is currently not installed. It is provided by
-the package '$toplevel.$attrs', which I will now install for you.
+the package '$toplevel#$attrs', which I will now install for you.
 EOF
-                nix-env -iA $toplevel.$attrs
+                nix shell $toplevel\#$attrs
                 if [ "$?" -eq 0 ]; then
                     $@ # TODO: handle pipes correctly if AUTO_RUN/INSTALL is possible
                     return $?
                 else
                     >&2 cat <<EOF
-Failed to install $toplevel.attrs.
+Failed to install $toplevel#attrs.
 $cmd: command not found
 EOF
                 fi
             elif ! [ -z "$NIX_AUTO_RUN" ]; then
-                nix-build --no-out-link -A $attrs "<$toplevel>"
+                nix build --no-link $toplevel\#$attrs
                 if [ "$?" -eq 0 ]; then
-                    # how nix-shell handles commands is weird
-                    # $(echo $@) is need to handle this
-                    nix-shell -p $attrs --run "$(echo $@)"
+                    nix shell $toplevel\#$attrs -c $@
                     return $?
                 else
                     >&2 cat <<EOF
-Failed to install $toplevel.attrs.
+Failed to install $toplevel#attrs.
 $cmd: command not found
 EOF
                 fi
@@ -70,7 +68,7 @@ EOF
                 >&2 cat <<EOF
 The program '$cmd' is currently not installed. You can install it
 by typing:
-  nix-env -iA $toplevel.$attrs
+  nix shell $toplevel#$attrs
 EOF
             fi
             ;;
@@ -83,7 +81,7 @@ EOF
             # ensure we get each element of attrs
             # in a cross platform way
             while read attr; do
-                >&2 echo "  nix-env -iA $toplevel.$attr"
+                >&2 echo "  nix shell $toplevel#$attr"
             done <<< "$attrs"
             ;;
     esac
