diff --git a/lua/base16-colorscheme.lua b/lua/base16-colorscheme.lua
index 5f4cc44..c3447be 100644
--- a/lua/base16-colorscheme.lua
+++ b/lua/base16-colorscheme.lua
@@ -97,19 +97,20 @@ function M.setup(colors)
     hi.NonText      = { guifg = M.colors.base03, guibg = nil,             gui = nil,    guisp = nil }
     hi.LineNr       = { guifg = M.colors.base04, guibg = M.colors.base00, gui = nil,    guisp = nil }
     hi.SignColumn   = { guifg = M.colors.base04, guibg = M.colors.base00, gui = nil,    guisp = nil }
-    hi.StatusLine   = { guifg = M.colors.base05, guibg = M.colors.base02, gui = 'none', guisp = nil }
+    hi.StatusLine   = { guifg = M.colors.base0B, guibg = M.colors.base02, gui = 'none', guisp = nil }
     hi.StatusLineNC = { guifg = M.colors.base04, guibg = M.colors.base01, gui = 'none', guisp = nil }
-    hi.VertSplit    = { guifg = M.colors.base05, guibg = M.colors.base00, gui = 'none', guisp = nil }
+    hi.VertSplit    = { guifg = M.colors.base01, guibg = M.colors.base00, gui = 'none', guisp = nil }
     hi.ColorColumn  = { guifg = nil,             guibg = M.colors.base01, gui = 'none', guisp = nil }
     hi.CursorColumn = { guifg = nil,             guibg = M.colors.base01, gui = 'none', guisp = nil }
-    hi.CursorLine   = { guifg = nil,             guibg = M.colors.base01, gui = 'none', guisp = nil }
-    hi.CursorLineNr = { guifg = M.colors.base04, guibg = M.colors.base01, gui = nil,    guisp = nil }
+    hi.CursorLine   = { guifg = nil,             guibg = M.colors.base02, gui = 'none', guisp = nil }
+    hi.CursorLineNr = { guifg = M.colors.base0B, guibg = M.colors.base01, gui = nil,    guisp = nil }
     hi.QuickFixLine = { guifg = nil,             guibg = M.colors.base01, gui = 'none', guisp = nil }
     hi.PMenu        = { guifg = M.colors.base05, guibg = M.colors.base01, gui = 'none', guisp = nil }
     hi.PMenuSel     = { guifg = M.colors.base01, guibg = M.colors.base05, gui = nil,    guisp = nil }
     hi.TabLine      = { guifg = M.colors.base03, guibg = M.colors.base01, gui = 'none', guisp = nil }
-    hi.TabLineFill  = { guifg = M.colors.base03, guibg = M.colors.base01, gui = 'none', guisp = nil }
+    hi.TabLineFill  = { guifg = M.colors.base03, guibg = M.colors.base02, gui = 'none', guisp = nil }
     hi.TabLineSel   = { guifg = M.colors.base0B, guibg = M.colors.base01, gui = 'none', guisp = nil }
+    hi.EndOfBuffer  = { guifg = M.colors.base00, guibg = nil,             gui = nil,    guisp = nil }
 
     -- Standard syntax highlighting
     hi.Boolean      = { guifg = M.colors.base09, guibg = nil,             gui = nil,    guisp = nil }
@@ -259,6 +260,38 @@ function M.setup(colors)
     hi.TSDefinitionUsage = { guifg = nil, guibg = nil, gui = 'underline', guisp = M.colors.base04 }
     hi.TSCurrentScope    = { guifg = nil, guibg = nil, gui = 'bold',      guisp = nil }
 
+    hi.IndentBlankLine = { guifg = M.colors.base01, guibg = nil, gui = 'none', guisp = nil }
+
+    -- Completion (nvim-cmp.lua)
+    hi.CmpItemAbbr =           { guifg = M.colors.base05, guibg = 'none', gui = 'none', guisp = nil }
+    hi.CmpItemAbbrDeprecated = { guifg = M.colors.base03, guibg = 'none', gui = 'none', guisp = nil }
+    hi.CmpItemAbbrMatch =      { guifg = M.colors.base05, guibg = 'none', gui = 'none', guisp = nil }
+    hi.CmpItemAbbrMatchFuzzy = { guifg = M.colors.base05, guibg = 'none', gui = 'none', guisp = nil }
+    hi.CmpItemKind =           { guifg = M.colors.base0C, guibg = 'none', gui = nil,    guisp = nil }
+    hi.CmpItemMenu =           { guifg = M.colors.base05, guibg = 'none', gui = 'none', guisp = nil }
+
+    -- BufferLine (barbar.nvim)
+    hi.BufferCurrent        = { guifg = M.colors.base0B, guibg = M.colors.base00, gui = nil, guisp = nil }
+    hi.BufferCurrentIndex   = { guifg = M.colors.base0B, guibg = M.colors.base00, gui = nil, guisp = nil }
+    hi.BufferCurrentMod     = { guifg = M.colors.base0E, guibg = M.colors.base00, gui = nil, guisp = nil }
+    hi.BufferCurrentSign    = { guifg = M.colors.base0B, guibg = M.colors.base00, gui = nil, guisp = nil }
+    hi.BufferCurrentTarget  = { guifg = M.colors.base08, guibg = M.colors.base00, gui = nil, guisp = nil }
+    hi.BufferCurrentIcon    = { guifg = nil,             guibg = M.colors.base00, gui = nil, guisp = nil }
+    hi.BufferVisible        = { guifg = M.colors.base0A, guibg = M.colors.base01, gui = nil, guisp = nil }
+    hi.BufferVisibleIndex   = { guifg = M.colors.base0A, guibg = M.colors.base01, gui = nil, guisp = nil }
+    hi.BufferVisibleMod     = { guifg = M.colors.base0E, guibg = M.colors.base01, gui = nil, guisp = nil }
+    hi.BufferVisibleSign    = { guifg = M.colors.base0A, guibg = M.colors.base01, gui = nil, guisp = nil }
+    hi.BufferVisibleTarget  = { guifg = M.colors.base08, guibg = M.colors.base01, gui = nil, guisp = nil }
+    hi.BufferVisibleIcon    = { guifg = nil,             guibg = M.colors.base01, gui = nil, guisp = nil }
+    hi.BufferInactive       = { guifg = M.colors.base04, guibg = M.colors.base02, gui = nil, guisp = nil }
+    hi.BufferInactiveIndex  = { guifg = M.colors.base05, guibg = M.colors.base02, gui = nil, guisp = nil }
+    hi.BufferInactiveMod    = { guifg = M.colors.base0E, guibg = M.colors.base02, gui = nil, guisp = nil }
+    hi.BufferInactiveSign   = { guifg = M.colors.base05, guibg = M.colors.base02, gui = nil, guisp = nil }
+    hi.BufferInactiveTarget = { guifg = M.colors.base08, guibg = M.colors.base02, gui = nil, guisp = nil }
+    hi.BufferInactiveIcon   = { guifg = nil,             guibg = M.colors.base02, gui = nil, guisp = nil }
+    hi.BufferTabpages       = { guifg = M.colors.base03, guibg = M.colors.base02, gui = nil, guisp = nil }
+    hi.BufferTabpageFill    = { guifg = M.colors.base03, guibg = M.colors.base02, gui = nil, guisp = nil }
+
     hi.NvimInternalError = { guifg = M.colors.base00, guibg = M.colors.base08, gui = 'none', guisp = nil }
 
     hi.NormalFloat  = { guifg = M.colors.base05, guibg = M.colors.base00, gui = nil,    guisp = nil }
