diff --git a/pfetch b/pfetch
index d47b878..79380da 100755
--- a/pfetch
+++ b/pfetch
@@ -1055,14 +1055,14 @@ get_wm() {
 }
 
 
-get_de() {
+get_desktop() {
     # This only supports Xorg related desktop environments though
     # this is fine as knowing the desktop environment on Windows,
     # macOS etc is useless (they'll always report the same value).
     #
     # Display the value of '$XDG_CURRENT_DESKTOP', if it's empty,
     # display the value of '$DESKTOP_SESSION'.
-    log de "${XDG_CURRENT_DESKTOP:-$DESKTOP_SESSION}" >&6
+    log desktop "${XDG_CURRENT_DESKTOP:-$DESKTOP_SESSION}" >&6
 }
 
 get_shell() {
@@ -1078,6 +1078,19 @@ get_editor() {
     log editor "${editor##*/}" >&6
 }
 
+get_term() {
+    # Display the value of '$TERMINAL', if it's empty, display the
+    # value of '$TERM'.
+    term=${TERMINAL:-"$TERM"}
+
+    log term "${term##*/}" >&6
+}
+
+get_scheme() {
+    # Display the '$SCHEME' environment variable.
+    log scheme "$(cat $HOME/.colorscheme)" >&6
+}
+
 get_palette() {
     # Print the first 8 terminal colors. This uses the existing
     # sequences to change text color with a sequence prepended
