diff --git a/plugin/number_toggle.vim b/plugin/number_toggle.vim
index d5dad54..80aa97e 100644
--- a/plugin/number_toggle.vim
+++ b/plugin/number_toggle.vim
@@ -4,6 +4,6 @@
 
 augroup numbertoggle
   autocmd!
-  autocmd BufEnter,FocusGained,InsertLeave,WinEnter * if &nu && mode() != "i" | set rnu   | endif
-  autocmd BufLeave,FocusLost,InsertEnter,WinLeave   * if &nu                  | set nornu | endif
+  autocmd BufEnter,FocusGained,InsertLeave,WinEnter,CmdlineLeave * if &nu && mode() != "i" | set rnu   | endif
+  autocmd BufLeave,FocusLost,InsertEnter,WinLeave,CmdlineEnter   * if &nu                  | set nornu | redraw | endif
 augroup END
