From 65171b8b67b8f45a057bdb8e08e04b26c853f1c6 Mon Sep 17 00:00:00 2001
From: Gabriel Fontes <hi@m7.rs>
Date: Tue, 7 Apr 2026 18:04:28 -0300
Subject: [PATCH]

---
 src/wl-paste.c | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/src/wl-paste.c b/src/wl-paste.c
index e359ad9..22f9df0 100644
--- a/src/wl-paste.c
+++ b/src/wl-paste.c
@@ -100,7 +100,7 @@ static struct types classify_offer_types(struct offer *offer) {
         ) {
             types.having_explicit_as_prefix = mime_type;
         }
-        if (strcmp(mime_type, x_kde_password_manager_hint) == 0) {
+        if (strcmp(mime_type, x_kde_password_manager_hint) == 0 || str_has_suffix(mime_type, "secret")) {
             /* We should be checking if it contains
              * the string "secret" as opposed to "public",
              * but for now let's just use the presence
--
2.52.0

