diff -r 1c32143a8460 src/wofi.c
--- a/src/wofi.c	Thu Oct 06 19:35:13 2022 -0700
+++ b/src/wofi.c	Fri Oct 21 21:10:01 2022 -0300
@@ -43,8 +43,6 @@

 #define PROTO_VERSION(v1, v2) (v1 < v2 ? v1 : v2)

-static const char* terminals[] = {"kitty", "termite", "alacritty", "foot", "gnome-terminal", "weston-terminal"};
-
 enum location {
 	LOCATION_CENTER,
 	LOCATION_TOP_LEFT,
@@ -914,13 +912,15 @@
 }

 void wofi_term_run(const char* cmd) {
-	if(terminal != NULL) {
-		execlp(terminal, terminal, "-e", cmd, NULL);
-	}
-	size_t term_count = sizeof(terminals) / sizeof(char*);
-	for(size_t count = 0; count < term_count; ++count) {
-		execlp(terminals[count], terminals[count], "-e", cmd, NULL);
-	}
+    char *shell = getenv("SHELL");
+    if (!shell) shell = "sh";
+
+    char *term = getenv("TERMINAL");
+    if (!term) term = terminal;
+
+	if(term != NULL) {
+		execlp(term, term, shell, "-i", "-c", cmd, NULL);
+ 	}
 	fprintf(stderr, "No terminal emulator found please set term in config or use --term\n");
 	exit(1);
 }
