diff -r 2cd009548707 src/wofi.c
--- a/src/wofi.c	Sun Dec 12 23:21:28 2021 -0800
+++ b/src/wofi.c	Thu Dec 23 07:05:31 2021 -0300
@@ -40,8 +40,6 @@
 #include <pango/pango.h>
 #include <gdk/gdkwayland.h>

-static const char* terminals[] = {"kitty", "termite", "alacritty", "gnome-terminal", "weston-terminal"};
-
 enum matching_mode {
 	MATCHING_MODE_CONTAINS,
 	MATCHING_MODE_FUZZY
@@ -915,13 +913,16 @@
 }

 void wofi_term_run(const char* cmd) {
-	if(terminal != NULL) {
-		execlp(terminal, terminal, "-e", cmd, NULL);
+    char *shell = getenv("SHELL");
+    if (!shell) shell = "sh";
+
+    char *term = getenv("TERMINAL");
+    if (!term) term = terminal;
+
+	if(term != NULL) {
+		execlp(term, term, shell, "-i", "-c", cmd, NULL);
 	}
-	size_t term_count = sizeof(terminals) / sizeof(char*);
-	for(size_t count = 0; count < term_count; ++count) {
-		execlp(terminals[count], terminals[count], "-e", cmd, NULL);
-	}
+
 	fprintf(stderr, "No terminal emulator found please set term in config or use --term\n");
 	exit(1);
 }
