diff --git a/scripts/xdg-open.in b/scripts/xdg-open.in
index feacc1f..32b79b2 100755
--- a/scripts/xdg-open.in
+++ b/scripts/xdg-open.in
@@ -378,7 +378,15 @@ search_desktop_file()
                 args=$(( args - 1 ))
             done
             [ $replaced -eq 1 ] || set -- "$@" "${target:-$target_uri}"
-            env "$command" "$@"
+            if [ x"$(get_key "${file}" "Terminal")" = x"true" ] && ( [ ! -t 0 ] || [ ! -t 1 ] ); then
+                if command -v xdg-terminal-exec &> /dev/null; then
+                    env xdg-terminal-exec env "$command" "$@"
+                else
+                    env xterm -e env "$command" "$@"
+                fi
+            else
+                env "$command" "$@"
+            fi
             exit_success
         fi
     fi
