diff --git a/lyrics/__init__.py b/lyrics/__init__.py
index da6fc52..d1d3329 100644
--- a/lyrics/__init__.py
+++ b/lyrics/__init__.py
@@ -5,15 +5,3 @@ CACHE_PATH = Path.home().joinpath('.cache', 'lyrics')
 CONFIG_PATH = Path.home().joinpath('.config', 'lyrics-in-terminal','lyrics.cfg')

 __version__ = '1.5.0'
-
-if not CONFIG_PATH.exists():
-    from shutil import copy
-    import os
-
-    dirname = Path(__file__).parent
-    src = dirname.joinpath('lyrics.cfg')
-
-    if not CONFIG_PATH.parent.exists():
-        os.makedirs(CONFIG_PATH.parent)
-
-    copy(src, CONFIG_PATH)
diff --git a/setup.py b/setup.py
index 10743d4..e7f0ded 100644
--- a/setup.py
+++ b/setup.py
@@ -2,7 +2,7 @@ from setuptools import setup, find_packages
 from setuptools.command.install import install
 import os

-from lyrics import __version__, CONFIG_PATH
+from lyrics import __version__

 this_directory = os.path.abspath(os.path.dirname(__file__))

@@ -10,44 +10,6 @@ with open(os.path.join(this_directory, 'README.md'), encoding='utf-8') as f:
     long_description = f.read()


-class PostInstallConfigUpdate(install):
-    """Post-installation for config update"""
-
-    def run(self):
-        install.run(self)
-        self.updateConfigFile()
-
-    def updateConfigFile(self):
-        if CONFIG_PATH.exists():
-            from configparser import ConfigParser
-
-            old_config = ConfigParser()
-            old_config.read(CONFIG_PATH)
-
-            new_config = ConfigParser()
-            new_config.read('lyrics/lyrics.cfg')
-
-            skip = True
-
-            for section in old_config.sections():
-                old_keys = {o for o in old_config[section]}
-                new_keys = {n for n in new_config[section]}
-
-                new_options = new_keys - old_keys
-
-                if len(new_options) == 0:
-                    continue
-                else:
-                    skip = False
-
-                for option in new_options:
-                    old_config[section][option] = new_config[section][option]
-
-            if not skip:
-                with open(CONFIG_PATH, 'w') as file:
-                    old_config.write(file)
-
-
 setup(
     name='lyrics-in-terminal',
     version=__version__,
@@ -77,8 +39,5 @@ setup(
         'full': ['python-mpd2']
     },
     python_requires='>=3.6',
-    cmdclass={
-        'install': PostInstallConfigUpdate
-    },
     include_package_data=True
 )
